pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -188,6 +208,183 @@ func handleInferenceGraphRawDeployment(ctx context.Context, cl client.Client, cl
+			if err != nil {
+				return errors.Wrap(err, "fails to apply auth-verifier service account for inference graph")
+			}
+		}
+
+		// Bind the required privileges to the Service Account
+		err = addAuthPrivilegesToGraphServiceAccount(ctx, clientset, graph)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := removeAuthPrivilegesFromGraphServiceAccount(ctx, clientset, graph)
+		if err != nil {
+			return err
+		}
+
+		err = deleteGraphServiceAccount(ctx, clientset, graph)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addAuthPrivilegesToGraphServiceAccount(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1.InferenceGraph) error {
+	clusterRoleBinding, err := clientset.RbacV1().ClusterRoleBindings().Get(ctx, constants.InferenceGraphAuthCRBName, metav1.GetOptions{})
+	if client.IgnoreNotFound(err) != nil {
+		return errors.Wrapf(err, "fails to get cluster role binding kserve-inferencegraph-auth-verifiers while configuring inference graph auth")
+	}
+
+	saName := graph.ResolveServiceAccountName()
+	if apierrors.IsNotFound(err) {
+		clusterRoleAuxiliary := rbacv1.ClusterRole{}
+		rbRoleRef := rbacv1cfg.RoleRef().
+			WithKind("ClusterRole").
+			WithName("system:auth-delegator").
+			WithAPIGroup(clusterRoleAuxiliary.GroupVersionKind().Group)
+		rbSubject := rbacv1cfg.Subject().
+			WithKind("ServiceAccount").
+			WithNamespace(graph.GetNamespace()).
+			WithName(saName)
+		crbApply := rbacv1cfg.ClusterRoleBinding(constants.InferenceGraphAuthCRBName).
+			WithRoleRef(rbRoleRef).
+			WithSubjects(rbSubject)
+
+		_, err = clientset.RbacV1().ClusterRoleBindings().Apply(ctx, crbApply, metav1.ApplyOptions{FieldManager: InferenceGraphControllerName})
+		if err != nil {
+			return errors.Wrapf(err, "fails to apply kserve-inferencegraph-auth-verifiers ClusterRoleBinding for inference graph")
+		}
+	} else {
+		isPresent := false
+		for _, subject := range clusterRoleBinding.Subjects {
+			if subject.Kind == "ServiceAccount" && subject.Name == saName && subject.Namespace == graph.GetNamespace() {
+				isPresent = true
+				break
+			}
+		}
+		if !isPresent {
+			clusterRoleBinding.Subjects = append(clusterRoleBinding.Subjects, rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      saName,
+				Namespace: graph.GetNamespace(),
+			})
+			_, err = clientset.RbacV1().ClusterRoleBindings().Update(ctx, clusterRoleBinding, metav1.UpdateOptions{FieldManager: InferenceGraphControllerName})
+			if err != nil {
+				return errors.Wrapf(err, "fails to bind privileges for auth verification to inference graph")
+			}
+		}
+	}
+
+	return nil
+}
+
+func removeAuthPrivilegesFromGraphServiceAccount(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1.InferenceGraph) error {
+	clusterRole, err := clientset.RbacV1().ClusterRoleBindings().Get(ctx, constants.InferenceGraphAuthCRBName, metav1.GetOptions{})
```

Comment:
Is clusterRole mandatory for IG?

Reply:
For checking authentication, yes. Because the TokenReview and SubjectAccessReview APIs are cluster-scoped.

Reply:
okay, thanks.